### PR TITLE
Centralize MODIFY processing behavior

### PR DESCRIPTION
**Pull request by thpr**
This prepares for other tokens that will want to process the same 3+association set of information, such as TEMPMODIFY

### CODE REVIEW

File: code/src/java/plugin/lsttokens/ModifyLst.java [hunk 1]
```diff
@@ -58,40 +56,71 @@ public String getTokenName()
 	@Override
 	public ParseResult parseNonEmptyToken(LoadContext context, VarHolder obj, String value)
 	{
-		//TODO These instanceof checks will fail - the VarHolder is a proxy :(
-		if (obj instanceof Ungranted)
+		/*
+		 * TODO Need to check the object type of the VarHolder to make sure it is legal.
```
Comment (javydreamercsw) [inline code/src/java/plugin/lsttokens/ModifyLst.java]: Not a big fan of TODO in source alone. This should be recorded in a JIRA issue so it doesn't fall thru the cracks and referenced here.

File: code/src/java/plugin/lsttokens/ModifyOtherLst.java [hunk 2]
```diff
@@ -63,16 +55,11 @@ public String getTokenName()
...
-		{
-			return new ParseResult.Fail(getTokenName() + " may not be used in Ungranted objects.");
-		}
-		if (obj instanceof Campaign)
-		{
-			return new ParseResult.Fail(
-				getTokenName() + " may not be used in Campaign Files.  " + "Please use the Global Modifier file");
-		}
+		/*
+		 * TODO Need to check the object type of the VarHolder to make sure it is legal.
```
Comment (javydreamercsw) [inline code/src/java/plugin/lsttokens/ModifyOtherLst.java]: Same TODO comment as before.

File: code/src/utest/plugin/lsttokens/ModifyLstTest.java [hunk 3]
```diff
@@ -202,6 +202,14 @@ public void testRoundRobinPriority() throws PersistenceLayerException
 		runRoundRobin("MyVar|ADD|3|PRIORITY=1090");
 	}
 
+	//TODO Ignore for now; reactivate later
```
Comment (javydreamercsw) [inline code/src/utest/plugin/lsttokens/ModifyLstTest.java]:
```suggestion
```

File: code/src/utest/plugin/lsttokens/ModifyLstTest.java [hunk 4]
```diff
@@ -202,6 +202,14 @@ public void testRoundRobinPriority() throws PersistenceLayerException
...
+//	@Test
```
Comment (javydreamercsw) [inline code/src/utest/plugin/lsttokens/ModifyLstTest.java]:
```suggestion
	@Test(enabled=false)
```

File: code/src/utest/plugin/lsttokens/ModifyLstTest.java [hunk 5]
```diff
@@ -202,6 +202,14 @@ public void testRoundRobinPriority() throws PersistenceLayerException
...
+//	public void testInvalidObject()
```
Comment (javydreamercsw) [inline code/src/utest/plugin/lsttokens/ModifyLstTest.java]:
```suggestion
	public void testInvalidObject()
```

File: code/src/utest/plugin/lsttokens/ModifyLstTest.java [hunk 6]
```diff
@@ -202,6 +202,14 @@ public void testRoundRobinPriority() throws PersistenceLayerException
...
+//	{
```
Comment (javydreamercsw) [inline code/src/utest/plugin/lsttokens/ModifyLstTest.java]:
```suggestion
	{
```

File: code/src/utest/plugin/lsttokens/ModifyLstTest.java [hunk 7]
```diff
@@ -202,6 +202,14 @@ public void testRoundRobinPriority() throws PersistenceLayerException
...
+//		assertFalse(token.parseToken(primaryContext, new Campaign(),
```
Comment (javydreamercsw) [inline code/src/utest/plugin/lsttokens/ModifyLstTest.java]:
```suggestion
		assertFalse(token.parseToken(primaryContext, new Campaign(),
```

File: code/src/utest/plugin/lsttokens/ModifyLstTest.java [hunk 8]
```diff
@@ -202,6 +202,14 @@ public void testRoundRobinPriority() throws PersistenceLayerException
...
+//				"MyVar|ADD|3").passed());
```
Comment (javydreamercsw) [inline code/src/utest/plugin/lsttokens/ModifyLstTest.java]:
```suggestion
				"MyVar|ADD|3").passed());
```

File: code/src/utest/plugin/lsttokens/ModifyLstTest.java [hunk 9]
```diff
@@ -202,6 +202,14 @@ public void testRoundRobinPriority() throws PersistenceLayerException
...
+//	}
```
Comment (javydreamercsw) [inline code/src/utest/plugin/lsttokens/ModifyLstTest.java]:
```suggestion
	}
```

File: code/src/utest/plugin/lsttokens/ModifyLstTest.java (hunk 3 above)
Comment (javydreamercsw) [reply code/src/utest/plugin/lsttokens/ModifyLstTest.java]: I suggest adding a comment on why is disabled and even better if it references a JIRA issue.

File: code/src/utest/plugin/lsttokens/ModifyLstTest.java (hunk 4 above)
Comment (thpr) [reply code/src/utest/plugin/lsttokens/ModifyLstTest.java]: This doesn't work for me and causes a compile error saying enabled is not a defined annotation for Test.  Is that a JUnit 5 thing?

Comment (javydreamercsw) [reply code/src/utest/plugin/lsttokens/ModifyLstTest.java]: I guess I confused with TestNG. Try @Disabled("Explicitly Disabled") instead.

Change the comment as needed.

**Pull request merged**